src/transports/smart_protocol.c

```diff
@@ -358,7 +314,10 @@ int git_smart__negotiate_fetch(git_transport *transport, git_repository *repo, c
 	if ((error = git_pkt_buffer_wants(wants, count, &t->caps, &data)) < 0)
 		return error;
 
-	if ((error = fetch_setup_walk(&walk, repo)) < 0)
+	if ((error = git_revwalk_new(&walk, repo)) < 0)
+		return error;
```

Comment:
`goto on_error` to clean up `data`?

Reply:
If you felt like it, making `data` the first param to `git_pkt_buffer_wants` might help in understanding that it's an out param...

Reply:
Fixed.

This is in the vicinity of #5105, hence I didn't want more extensive changes here. If that's fine with you I'll tackle that out there.